src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1123,6 +1123,35 @@ private CodeParameter GetIndexerParameter(OpenApiUrlTreeNode currentNode)
         };
         return result;
     }
+
+    private CodeType? GetEnumType(OpenApiUrlTreeNode currentNode, IOpenApiParameter parameter)
+    {
+        IOpenApiSchema? enumCandidateSchema = parameter.Schema;
+        // many specs wrap refs under allOf: [ { $ref: ... } ]
+        if (enumCandidateSchema?.AllOf is { Count: 1 } && enumCandidateSchema.AllOf.FirstOrDefault() is IOpenApiSchema singleAllOf)
+            enumCandidateSchema = singleAllOf;
+
+        if (enumCandidateSchema is null || modelsNamespace is null)
```

Comment:
There's an extension method FlattenIfRequired to handle such cases. Please call that method instead.